lib/ansible/modules/cloud/vmware/vmware_spbm.py

```diff
@@ -0,0 +1,164 @@
+            module.fail_json(msg=to_native(e))
+
+    def force_provision(self):
+        """Allow Force Provision on vsanDatastore"""
+        datastore = find_datastore_by_name(self.content, "vsanDatastore")
+        vsan_profile_id = self.spbmclient.get_ds_default_profile(datastore)
+        rules = [{'forceProvisioning': True}]
+        self.spbmclient.update_storage_profile(profile_id=vsan_profile_id, rules=rules)
+
+    def process_state(self):
+        """Process Current State of Profiles"""
+        profiles = self.spbmclient.get_profiles()
+        if not profiles:
+            self.module.fail_json(changed=False, msg="Could not retrieve \
+            Storage Profile Information from vCenter")
+        else:
+            for profile in profiles:
+                if self.module.params['profile_name'] == profile.name:
+                    self.currentstate = 'present'
+        # Current State of System and Desired State are same
+        if self.currentstate == self.desiredstate:
+            self.module.exit_json(changed=False)
+
+        # Profile is absent and you need to create one
+        if self.currentstate == 'absent' and self.desiredstate == 'present':
+            profileId = self.spbmclient.create_storage_profile(profile_name=self.name, \
+            description=self.description, rules=self.rules)
+            if profileId:
+                self.module.exit_json(changed=True, msg="Created Storage Profile {} \
+                and ID {}".format(self.name, profileId.uniqueId))
+            else:
+                self.module.fail_json(changed=False, msg="Failed to create \
+                Storage Profile {}".format(self.name))
+
+        # Profile is Present and you need to delete one
+        if self.currentstate == 'present' and self.desiredstate == 'absent':
+            results = self.spbmclient.delete_storage_profile(profile_name=self.name)
+            if not results:
+                self.module.exit_json(changed=True, \
+                msg="Deleted Storage Profile {}".format(self.name))
+            else:
+                profile_op_outcome = results[0]
+                if profile_op_outcome.fault is not None:
+                    self.module.fail_json(changed=False, msg=str(profile_op_outcome.fault))
+
+
+def main():
+    """Manages VM Storage Profiles"""
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(dict(profile_name=dict(required=True, type='str'), \
+    description=dict(required=False, type='str', default='Sample Storage Profile'), \
+    force_provision=dict(required=False, type='bool', default=False), \
+    rules=dict(required=False, type='list', default=[{'stripeWidth': 1}, \
+    {'hostFailuresToTolerate': 1}, {'replicaPreference':'RAID-5/6 (Erasure Coding) - Capacity'}]), \
+    state=dict(default='present', choices=['present', 'absent'], type='str')))
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=False)
+    if not HAS_PYVMOMI:
```

Comment:
When using the general PyVmomi class from the VMware utils, you don't need this.